internal/storage/db/internal/db.go

```diff
@@ -48,9 +50,71 @@ type dbStorage struct {
 	*storage.SubscriptionManager
 }
 
+func (s *dbStorage) AddOrUpdateSchema(ctx context.Context, sch *schemav1.Schema) error {
+	if sch == nil {
+		return fmt.Errorf("schema cannot be nil")
+	}
+
+	schemaRecord := Schema{
+		ID:          SchemaDefaultID,
+		Description: sch.Description,
+		Disabled:    sch.Disabled,
+		Definition:  SchemaDefWrapper{sch},
+	}
+
+	_, err := s.db.From(SchemaTbl).
+		Where(goqu.C(SchemaTblIDCol).Eq(SchemaDefaultID)).
+		Delete().
+		Executor().
+		ExecContext(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to delete the schema before inserting: %w", err)
+	}
+
+	if _, err := s.db.Insert(SchemaTbl).
+		Rows(schemaRecord).
+		Executor().
+		ExecContext(ctx); err != nil {
+		return fmt.Errorf("failed to insert the schema: %w", err)
+	}
```

Comment:
Since there's only one record, can't we simply do an `UPSERT` instead? See `AddOrUpdate` method.